mhkit/loads/general.py

```diff
@@ -165,3 +165,53 @@ def damage_equivalent_load(data_signal, m, bin_num=100, data_length=600):
     DEL = DELs.sum() ** (1/m)
 
     return DEL
+
+def cumulative_rainflow_spectrum(data_signal, m, bin_num=100, data_length=600):
+    '''
+    Calculates the cumulative rainflow spectrum for power production for each load quantity.
+    Summing all the individual rain flow cycle counts of each file in the power production
+    capture matrix, based on IEC TS 62600-3:2020 ED1. 4-point rainflow counting algorithm from
+    fatpack module is based on the following resources:
+
+    - `C. Amzallag et. al. Standardization of the rainflow counting method for
+      fatigue analysis. International Journal of Fatigue, 16 (1994) 287-293`
+    - `ISO 12110-2, Metallic materials - Fatigue testing - Variable amplitude
+      fatigue testing.`
+    - `G. Marsh et. al. Review and application of Rainflow residue processing
+      techniques for accurate fatigue damage estimation. International Journal
+      of Fatigue, 82 (2016) 757-765`
+
+
+    Parameters:
+    -----------
+    data_signal : 2d array
+        Data signals being analyzed
+    bin_num : int
+        Number of bins for rainflow counting method (minimum=100)
```

Comment:
I would add an assert statement checking to see if this value is above 100

Reply:
Hey @rpauly18 we are no longer using assert statements. See #242.

For the current assert the replacement would be:
```
    if not isinstance(bin_num, (float, int)):
        raise ValueError('bin_num must be of type float or int')
```

ChatGPT is pretty good at converting these but we need to always check its work.